### PR TITLE
Introduce builtin identity handling for ambient Io

### PR DESCRIPTION
**Pull request by Kroisse**
## Summary
- Distinguish compiler-owned builtin abilities from source-defined abilities with explicit origin metadata
- Register `std::io::Io` as a virtual builtin binding so imports, completions, and hover/definition tooling can recognize it without synthetic AST declarations
- Treat builtin `Io` as ambient in type checking and main-function effect validation, while still rejecting user handlers for it
- Update resolution to prefer local declarations over builtin imports, and align ability construction helpers across the front end

## Testing
- Added and updated unit tests for builtin/source ability identity, resolver behavior, effect conversion, and LSP completion/indexing
- Verified existing tests continue to cover calling-convention and typeck paths for ability handling

## Summary by CodeRabbit

* **New Features**
  * Added compiler-provided ambient I/O support via `std::io::Io` (as a virtual built-in, not a source-declared ability).
  * Added editor completion and hover details for the `std::io::Io` ability.
* **Bug Fixes**
  * Improved name resolution so local definitions correctly shadow built-ins and imports.
  * Updated typechecking diagnostics for handled/unhandled effects involving ambient `std::io::Io`.
* **Documentation**
  * Documented the virtual built-in binding model, identity/resolution rules, and editor behavior for `std::io::Io`.
* **Tests**
  * Expanded end-to-end and editor coverage, including builtin-vs-user `Io` edge cases and a new WASM compilation test.

### CODE REVIEW

File: crates/tribute-front/src/typeck/checker/expr.rs [hunk 1]
```diff
@@ -2161,7 +2175,7 @@ impl<'db> TypeChecker<'db> {
             ResolvedRef::TypeDef { id } => {
                 // TypeDef might be an ability reference in handler context
                 // Create an AbilityId with the same qualified name
-                Some(AbilityId::new(self.db(), id.qualified(self.db())))
+                Some(AbilityId::source(self.db(), id.qualified(self.db())))
```
[the remark this answers is not in the document]
Comment (Kroisse) [reply crates/tribute-front/src/typeck/checker/expr.rs:2178]: Thanks. I checked this against the current resolution path and am leaving this branch unchanged. Compiler-owned Io is injected as Binding::Ability and converted directly to ResolvedRef::Ability { id }, so extract_ability_id_from_ref returns the stored AbilityId with its Builtin origin intact. ResolvedRef::TypeDef is produced for source type definitions, not builtin abilities. Using from_resolved_path in the TypeDef fallback would instead classify a source type whose qualified spelling happens to be std::io::Io as builtin, which would violate the source/builtin origin separation this PR introduces. The existing test_builtin_io_cannot_be_handled end-to-end test confirms that builtin Io reaches is_builtin_io and emits the ambient-handler diagnostic. Therefore the TypeDef representation assumed by this finding does not occur in the current resolver.
